plasma_framework/migrations/1041_setup_vault.js

```diff
@@ -0,0 +1,179 @@
+const EthDepositVerifier = artifacts.require('EthDepositVerifier');
+const Erc20DepositVerifier = artifacts.require('Erc20DepositVerifier');
+const util = require('util');
+const childProcess = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const config = require('../config.js');
+const pck = require('../package.json');
+
+const expectedBlockTime = 1000;
+/* eslint-disable arrow-body-style */
+const sleep = (milliseconds) => {
+    return new Promise(resolve => setTimeout(resolve, milliseconds));
+};
+
+module.exports = async (
+    deployer,
+    _,
+    // eslint-disable-next-line no-unused-vars
+    [deployerAddress, maintainerAddress, authorityAddress],
+) => {
+    const vault = process.env.VAULT || false;
+    if (vault) {
+        console.log('Performing final setup with multisig maintainer');
+        const plasmaFramework = await PlasmaFramework.deployed();
+        const sha = childProcess.execSync('git rev-parse HEAD').toString().trim().substring(0, 7);
+        const ethDepositVerifier = await EthDepositVerifier.deployed();
+        const ethVault = await EthVault.deployed();
+        const erc20DepositVerifier = await Erc20DepositVerifier.deployed();
+        const erc20Vault = await Erc20Vault.deployed();
+        const MORE_VP = config.frameworks.protocols.moreVp;
+        const PAYMENT_TX_TYPE = config.registerKeys.txTypes.payment;
+        const FEE_TX_TYPE = config.registerKeys.txTypes.fee;
+        const paymentExitGame = await PaymentExitGame.deployed();
+        const feeExitGame = await FeeExitGame.deployed();
+        const buildDir = path.resolve(__dirname, '../../MultiSigWallet/build/multisig_instance');
+        const gnosisMultisigAddress = fs.readFileSync(buildDir, 'utf8');
+        const gnosisMultisigAbi = { constant: false,
+            inputs: [
+                { name: 'destination', type: 'address' },
+                { name: 'value', type: 'uint256' },
+                { name: 'data', type: 'bytes' },
+            ],
+            name: 'submitTransaction',
+            outputs: [{ name: 'transactionId', type: 'uint256' }],
+            payable: false,
+            type: 'function',
+            signature: '0xc6427474' };
+        // ethVault.setDepositVerifier
+        const setDepositVerifier = web3.eth.abi.encodeFunctionCall(ethVault.abi.find(o => o.name === 'setDepositVerifier'), [ethDepositVerifier.address]);
+        const gnosisSetDepositVerifier = web3.eth.abi.encodeFunctionCall(gnosisMultisigAbi, [ethVault.address, 0, setDepositVerifier]);
+        let transaction = await web3.eth.sendTransaction({ gas: 3000000, to: gnosisMultisigAddress, from: deployerAddress, data: gnosisSetDepositVerifier });
+        console.log(`Submitted transaction with hash for ETH setDepositVerifier: ${transaction.transactionHash}`);
+        let transactionReceipt = null;
+        while (transactionReceipt == null) { // Waiting expectedBlockTime until the transaction is mined
```

Comment:
I believe js prefer `===` instead of `==` for stricter comparison though I forgot what exactly the difference is 😅 But in general it is better to forgot `==` exists.